docs/en/guides/pipeline-integrations/administration/apps-and-engines-config-reference.md

```diff
@@ -6,3 +6,97 @@ lang: en
+_This document describes functionality only available if you have taken control over a Toolkit configuration. For more info, see  [{% include product %} Integrations Admin Guide](https://support.shotgunsoftware.com/hc/en-us/articles/115000067493)._
+
+# Introduction
+
+This document contains specifications for the various file formats that Sgtk uses for its configuration and settings. Please note that this is a reference document which outlines all the various options and parameters available. For best practices on how to manage your configuration, please see the following document:
+
+[> Configuration Management best practices.](https://support.shotgunsoftware.com/hc/en-us/articles/219033168)
+
+# {% include product %} Pipeline Toolkit Environments
+
+Three major components exists in Toolkit:
+
+-   _An engine_  provides a translation layer or an adapter between a host application (such as Maya or Nuke) and Sgtk Apps. Apps typically use python and PySide, and it is the responsibility of the engine to present the host application in a standardized fashion and for example add pyside on top of the host application if this doesn't exist already.
+    
+-   _An app_  provides a piece of business logic, it is essentially a tool that does something. Apps can be hand crafted to work in a specific host application, or they can be designed to run in more than one host application.
+    
+-   _A framework_  is a library which can be used by engines, apps or other frameworks. A framework makes it possible to more easily manage code or behaviour which is shared between multiple apps.
+    
+
+An  _environment file_  contains the configuration settings for a collection of engines, apps and frameworks. Such a collection is called an Environment. Sgtk launches different environments for different files or different people. You can for example have an environment for Shot production and environment for Rigging. Each environment is a single yaml file.
+
+Environment files are located at  `/<sgtk_root>/software/shotgun/<project_name>/config/env`
+
+The yaml file has the following basic format:
+
+    engines:
+        tk-maya:
+            location
+            engine settings
+    
+            apps:
+                tk-maya-publish:
+                    location
+                    app settings
+    
+                tk-maya-revolver:
+                    location
+                    app settings
+    
+        tk-nuke:
+            location
+            engine settings
+    
+            apps:
+                tk-nuke-setframerange:
+                    location
+                    app settings
+    
+                tk-nuke-nukepub:
+                    location
+                    app settings
+    
+    frameworks:
+        tk-framework-tools:
+            location
+            framework settings
```

Comment:
Let's do this as a code block so we can get syntax highlighting:

```yaml
foo:
    bar
```

I can't do it with a suggestion here in the PR because the backticks for the code block can't be properly escaped, but it's a quick thing for you to do on your system and push.